website/docs/r/vdc.html.markdown

```diff
@@ -0,0 +1,124 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vdc"
+sidebar_current: "docs-vcd-resource-vdc"
+description: |-
+  Provides a vCloud Director Provider VDC resource. This can be used to create and delete an VDC.
+---
+
+# vcd\_vdc
+
+Provides a vCloud Director VDC resource. This can be used to create and delete a VDC.
+Requires system administrator privileges.
+
+Supported in provider *v2.2+*
+
+## Example Usage
+
+```hcl
+provider "vcd" {
+  user     = "${var.admin_user}"
+  password = "${var.admin_password}"
+  vdc      = "System"
+  url      = "https://AcmeVcd/api"
+}
+
+resource "vcd_vdc" "my-vdc" {
+  name        = "my-vdc"
+  description = "The pride of my work"
+  org         = "my-org"
+
+  allocation_model = "ReservationPool"
+  network_pool_name = "vDC1-VXLAN-NP"
+  provider_vdc_name = "vDC1"
+
+  compute_capacity {
+    cpu {
+      units     = "MHz"
+      allocated = 2048
+      limit     = 2048
+      reserved  = 2048
+      used      = 0
+      overhead  = 0
+    }
+
+    memory {
+      units     = "MB"
+      allocated = 2048
+      limit     = 2048
+      reserved  = 2048
+      used      = 0
+      overhead  = 0
+    }
+  }
+
+  storage_profile {
+    name     = "storage-name"
+    enabled  = true
+    units    = "MB"
+    limit    = 10240
+    default  = true    
+  }
+
+  is_enabled             = true
+  is_thin_provision      = true
+  uses_fast_provisioning = true
+  delete_force           = true
+  delete_recursive       = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional; *v2.0+*) Organization to create the VDC in, optional if defined at provider level
+* `name` - (Required) VDC name
+* `description` - (Optional) VDC friendly description
+* `allocation_model` - (Required) The allocation model used by this VDC; must be one of {AllocationVApp, AllocationPool, ReservationPool}
+* `compute_capacity` - (Required) The compute capacity allocated to this VDC.  See [Compute Capacity](#computecapacity) below for details.
+* `nic_quota` - (Optional) Maximum number of virtual NICs allowed in this VDC. Defaults to 0, which specifies an unlimited number.
+* `network_quota` - (Optional) Maximum number of network objects that can be deployed in this VDC. Defaults to 0, which means no networks can be deployed.
```

Comment:
Just double checking, does 0 really mean no networks as opposed to unlimited?

Reply:
This message is copied from here https://code.vmware.com/apis/287/vcloud/doc/doc/types/CreateVdcParamsType.html

Reply:
In UI you can provide more than 0 when choose network pool